Source/WebCore/Modules/plugins/YouTubePluginReplacement.h

```diff
@@ -37,28 +38,28 @@ class YouTubePluginReplacement final : public PluginReplacement {
 public:
     static void registerPluginReplacement(PluginReplacementRegistrar);
 
-    typedef HashMap<String, String> KeyValueMap;
+    using KeyValueMap = HashMap<String, String>;
 
-    WEBCORE_EXPORT static String youTubeURLFromAbsoluteURL(const URL& srcURL, const String& srcString);
+    WEBCORE_EXPORT static AtomString youTubeURLFromAbsoluteURL(const URL& srcURL, const AtomString& srcString);
 
 private:
-    YouTubePluginReplacement(HTMLPlugInElement&, const Vector<String>& paramNames, const Vector<String>& paramValues);
-    static Ref<PluginReplacement> create(HTMLPlugInElement&, const Vector<String>& paramNames, const Vector<String>& paramValues);
+    YouTubePluginReplacement(HTMLPlugInElement&, const Vector<AtomString>& paramNames, const Vector<AtomString>& paramValues);
+    static Ref<PluginReplacement> create(HTMLPlugInElement&, const Vector<AtomString>& paramNames, const Vector<AtomString>& paramValues);
     static bool supportsMIMEType(const String&);
     static bool supportsFileExtension(StringView);
     static bool supportsURL(const URL&);
     static bool isEnabledBySettings(const Settings&);
 
     void installReplacement(ShadowRoot&) final;
 
-    String youTubeURL(const String& rawURL);
+    AtomString youTubeURL(const AtomString& rawURL);
 
     bool willCreateRenderer() final { return m_embedShadowElement; }
     RenderPtr<RenderElement> createElementRenderer(HTMLPlugInElement&, RenderStyle&&, const RenderTreePosition&) final;
 
     WeakPtr<HTMLPlugInElement> m_parentElement;
     RefPtr<YouTubeEmbedShadowElement> m_embedShadowElement;
-    KeyValueMap m_attributes;
+    HashMap<AtomString, AtomString> m_attributes;
```

Comment:
I have some comments about this, but I am pretty sure that plug-in replacement is obsolete code that can be deleted instead of being carefully updated. We should check into that. I don’t think it makes too much sense to have a `KeyValueMap` type as a member of the class that we don’t use here.

Reply:
Oh, this code is actually used. We tried to drop it and then we had to add it back because it broke some sites. I will look into the KeyValueMap though.